### DISCUSSION TITLE
Accept comma as a decimal separator

### DISCUSSION THREAD

**Discussion by Openuser87** [sysadminsmedia/homebox#719] [Ideas]
### First Check

- [x] This is not a feature request
- [x] I added a very descriptive title to this issue.
- [x] I used the GitHub search to find a similar issue and didn't find it.
- [x] I searched the documentation, with the integrated search.
- [x] I already read the docs and didn't find an answer.
- [x] I can replicate the issue inside the Demo install.

### Homebox Version

0.15.2

### What is the issue you are experiencing?

When entering purchase price with comma (e.g. `19,99`), the value is ignored after saving (price is `0`). Would be great to accept / handle such values (have a setting for it if necessary), as this is a wide-spread standard for decimal separator in European countries.

### How can the maintainer reproduce the issue?

- create a new item
- enter e.g. `19,99` as purchase price
- save

### Deployment

Docker (Linux)

### OS Architechture

ARM64

**Comment by tankerkiller125** [comment]
This is a limitation of the float field type and the number input on the browser.

Attempting to accommodate this would be an interestingly difficult task given that some countries use commas for thousand place separators, and other use it for the change. While other countries have the inverse.

Right now, as far as I can tell and while I was trying to sort other things, there is no good way to handle this, and make sure that it properly gets converted to a float (which mandates the `1245.12` format)

I should note however, that I think your browser cache may not be cleared, as the input field should not be accepting commas at all as of 0.15.2 (validated on demo), and if you did save it without noticing the comma wasn't accepted it should still save, although as 1999 instead of 19.99

Please see #204 for some info on the comma thing previously brought up.

**Comment by Openuser87** [comment]
> Right now, as far as I can tell and while I was trying to sort other things, there is no good way to handle this, and make sure that it properly gets converted to a float (which mandates the 1245.12 format)

Couldn't we have a setting in user's profile, and then apply some parsing on the field value according to that setting?

> I should note however, that I think your browser cache may not be cleared, as the input field should not be accepting commas at all as of 0.15.2 (validated on demo), and if you did save it without noticing the comma wasn't accepted it should still save, although as 1999 instead of 19.99

Just tested again (purchase price 19,99) in a private window on Firefox at https://demo.homebox.software/ - I still get no feedback from the field about any error, and the price is 0 after saving...

> Please see https://github.com/sysadminsmedia/homebox/issues/204 for some info on the comma thing previously brought up.

Sorry, didn't find this one (probably searched only open issues...). The issue is slightly different though...

**Comment by tankerkiller125** [comment]
Please note we're converting this issue into a discussion as per our new procedures/policies. You can read more about that in our announcement #712
